meta-resin-common/classes/image_types_resin.bbclass

```diff
@@ -1,183 +1,213 @@
-
-	# After creating the extended partition the next logical parition needs a IMAGE_ROOTFS_ALIGNMENT in front of it
-	START=$(expr ${START} \+ ${IMAGE_ROOTFS_ALIGNMENT})
-	END=$(expr ${START} \+ ${CONFIG_SIZE_ALIGNED})
-    parted -s ${RESIN_SDIMG} unit KiB mkpart logical fat16 ${START} ${END}
-
-	# Create Data (ext4) partition
-	START=$(expr ${END} \+ ${IMAGE_ROOTFS_ALIGNMENT})
-	parted -s ${RESIN_SDIMG} -- unit KiB mkpart logical ext4 ${START} -1s
-
-	# Create a vfat filesystem with boot files
-	BOOT_BLOCKS=$(LC_ALL=C parted -s ${RESIN_SDIMG} unit b print | awk '/ 1 / { print substr($4, 1, length($4 -1)) / 512 /2 }')
-	rm -rf ${WORKDIR}/boot.img
-	mkfs.vfat -n "${RESIN_BOOT_FS_LABEL}" -S 512 -C ${WORKDIR}/boot.img $BOOT_BLOCKS
-	echo "Copying files in RESIN_BOOT_PARTITION_FILE"
-    echo -n '' > ${WORKDIR}/${RESIN_BOOT_FS_LABEL}.${FINGERPRINT_EXT}
-    for RESIN_BOOT_PARTITION_FILE in ${RESIN_BOOT_PARTITION_FILES}; do
+    # resin-state
+    RESIN_STATE_SIZE_ALIGNED=$(expr ${RESIN_STATE_SIZE} \+ ${RESIN_IMAGE_ALIGNMENT} \- 1)
+    RESIN_STATE_SIZE_ALIGNED=$(expr ${RESIN_STATE_SIZE_ALIGNED} \- ${RESIN_STATE_SIZE_ALIGNED} \% ${RESIN_IMAGE_ALIGNMENT})
+
+    # resin-data
+    if [ -n "${RESIN_DATA_FS}" ]; then
+        RESIN_DATA_SIZE=`du -bks ${RESIN_DATA_FS} | awk '{print $1}'`
+        RESIN_DATA_SIZE_ALIGNED=$(expr ${RESIN_DATA_SIZE} \+ ${RESIN_IMAGE_ALIGNMENT} \- 1)
+        RESIN_DATA_SIZE_ALIGNED=$(expr ${RESIN_DATA_SIZE_ALIGNED} \- ${RESIN_DATA_SIZE_ALIGNED} \% ${RESIN_IMAGE_ALIGNMENT})
+    else
+        RESIN_DATA_SIZE_ALIGNED=${RESIN_IMAGE_ALIGNMENT}
+    fi
 
+    # Total raw image size
+    RESIN_RAW_IMG_SIZE=$(expr \
+        ${RESIN_IMAGE_ALIGNMENT} \+ \
+        ${RESIN_BOOT_SIZE_ALIGNED} \+ \
+        ${RESIN_ROOTA_SIZE_ALIGNED} \+ \
+        ${RESIN_ROOTB_SIZE_ALIGNED} \+ \
+        ${RESIN_IMAGE_ALIGNMENT} \+ \
+        ${RESIN_STATE_SIZE_ALIGNED} \+ \
+        ${RESIN_IMAGE_ALIGNMENT} \+ \
+        ${RESIN_DATA_SIZE_ALIGNED} \
+    )
+    echo "Creating raw image as it follow:"
+    echo "  Boot partition ${RESIN_BOOT_SIZE_ALIGNED} KiB [$(expr ${RESIN_BOOT_SIZE_ALIGNED} \/ 1024) MiB]"
+    echo "  Root A partition ${RESIN_ROOTA_SIZE_ALIGNED} KiB [$(expr ${RESIN_ROOTA_SIZE_ALIGNED} \/ 1024) MiB]"
+    echo "  Root B partition ${RESIN_ROOTA_SIZE_ALIGNED} KiB [$(expr ${RESIN_ROOTB_SIZE_ALIGNED} \/ 1024) MiB]"
+    echo "  State partition ${RESIN_STATE_SIZE_ALIGNED} KiB [$(expr ${RESIN_STATE_SIZE_ALIGNED} \/ 1024) MiB]"
+    echo "  Data partition ${RESIN_DATA_SIZE_ALIGNED} KiB [$(expr ${RESIN_DATA_SIZE_ALIGNED} \/ 1024) MiB]"
+    echo "---"
+    echo "Total raw image size ${RESIN_RAW_IMG_SIZE} KiB [$(expr ${RESIN_RAW_IMG_SIZE} \/ 1024) MiB]"
+
+    #
+    # Generate the raw image with partition table
+    #
+
+    dd if=/dev/zero of=${RESIN_RAW_IMG} bs=1024 count=0 seek=${RESIN_RAW_IMG_SIZE}
+    parted -s ${RESIN_RAW_IMG} mklabel msdos
+
+    # resin-boot
+    START=${IMAGE_ROOTFS_ALIGNMENT}
```

Comment:
Should we not use RESIN_IMAGE_ALIGNMENT here instead? Everywhere else RESIN_IMAGE_ALIGNMENT is used, it's just here that IMAGE_ROOTFS_ALIGNMENT is still used